### ISSUE TITLE
Seeing "Form Mixin requires component to be nested in a Form"

### ISSUE THREAD

**Issue by binarykitchen** [christianalfoni/formsy-react#110]
but no idea why. This with the latest version. Any clues?

**Comment by christianalfoni** [comment]
You have to wrap all your form elements in a Formsy.Form:

```js
<Formsy.Form>
  <MyInputElement/>
</Formsy.Form>
```

**Comment by christianalfoni** [comment]
I can improve that error message to be more explicit :-)

**Comment by binarykitchen** [comment]
Well, I already do, sort of, but nested:

```jsx
<Formsy.Form>
  <Row>
    <MyInputElement/>
  </Row>
</Formsy.Form>
```

is this supported?

**Comment by christianalfoni** [comment]
This is supported yes, but not when the Form is located in a different component

**Comment by firstdoit** [comment]
@binarykitchen I had the same problem. I needed to have a form with different inputs depending on country:

```
<Formsy.Form>
  {countryInputs}
</Formsy.Form>
```

Insteads of making `countryInputs` a component, I simply defined a function that returns arbitrary elements. Formsy is happy with that because the `Formsy` and the `Inputs` are still on the same render.

https://github.com/vtex/react-address-book/blob/f0eb8f1e111e3e286c7ef5378ad12c191f53b6c6/src/components/input-groups/InputGroupBRA.js

BTW, thanks for this awesome lib, @christianalfoni :)
